### PR TITLE
Add new configuration to the max limit of scim2/users endpoint

### PR DESCRIPTION
**Pull request by sandushi**
### Proposed changes in this pull request

- $subject
- The config is enabled by default for the master fix. To disable the config, add the below config to the deployment.toml.
- This is done to use the server wide max limit for the scim2/Users Search endpoint response

```
[scim2
consider_server_wide_user_endpoint_max_limit]=false
```

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [x] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java [hunk 1]
```diff
@@ -1873,4 +1873,20 @@ public static boolean isSCIM2UserMaxItemsPerPageEnabled() {
         }
         return Boolean.parseBoolean(scim2UserMaxItemsPerPageEnabledProperty);
     }
+
+    /**
+     * Read the SCIM User Endpoint Consider Server Wide config and returns it.
+     *
+     * @return If SCIM User Endpoint Consider Server Wise Config is enabled.
```
Comment (pavinduLakshan) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:1880]:
```suggestion
     * @return whether the `ConsiderServerWideUserEndpointMaxLimitEnabled` config for the SCIM User Endpoint  is enabled.
```

Comment (sandushi) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:1969]: I have followed the same return comment as previous methods

**Review by sadilchamishka — APPROVED** [review]

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java [hunk 2]
```diff
@@ -1962,4 +1962,20 @@ public static boolean isSCIM2UserMaxItemsPerPageEnabled() {
...
+     */
+    public static boolean isConsiderServerWideUserEndpointMaxLimitEnabled() {
```
Comment (sadilchamishka) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:1971]: We can have the method in here `https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/master/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java` 

We can move to the framework identityUtil if it is identity related concern.

Comment (sandushi) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:1971]: will do

Comment (sadilchamishka) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:1971]: or else any commonly used methods are moved to here. As this methods looks useful in scim user manager, we can have the method implemented in that level.

**Review by sadilchamishka — APPROVED** [review]

**Pull request merged**